kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationIT.java

```diff
@@ -0,0 +1,282 @@
+
+        int port = OAUTH_ENDPOINT_URL.getPort();
+        oauthServer = new OauthServerContainer(OauthBearerValidationIT.DOCKER_IMAGE_NAME);
+        oauthServer.setWaitStrategy(new LogMessageWaitStrategy().withRegEx(".*started server on address.*"));
+        oauthServer.addFixedExposedPort(port, port);
+        oauthServer.withEnv("SERVER_PORT", port + "");
+        oauthServer.start();
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        if (oauthServer != null) {
+            oauthServer.close();
+        }
+    }
+
+    @AfterEach
+    public void afterEach() throws Exception {
+        // https://issues.apache.org/jira/browse/KAFKA-17134
+        // Workaround for defect in Kafka where closed VerificationKeyResolvers get left in the
+        // cache. This is impactful to the proxy because its config is constant (so cache hits).
+        // The reason kafka broker doesn't suffer this itself is because the config is different between
+        // clusters instances (port numbers, log dir etc. are different).
+        var cacheField = OAuthBearerValidatorCallbackHandler.class.getDeclaredField("VERIFICATION_KEY_RESOLVER_CACHE");
+        cacheField.setAccessible(true);
+        ((Map<?, ?>) cacheField.get(null)).clear();
+    }
+
+    @Test
+    void successfulAuthWithValidToken() {
+        var config = getClientConfig(TOKEN_ENDPOINT_URL);
+
+        try (var tester = kroxyliciousTester(getConfiguredProxyBuilder());
+                var admin = tester.admin(config)) {
+            performClusterOperation(admin);
+
+            var text = AdminHttpClient.INSTANCE.getFromAdminEndpoint(METRICS).body();
+            var allMetrics = SimpleMetric.parse(text);
+
+            var saslHandshakeRequestsGoingUpCount = findFirstMetricMatching(allMetrics, UPSTREAM_SASL_HANDSHAKE_LABELS_PREDICATE);
+            var saslAuthenticationResponsesComingDown = findFirstMetricMatching(allMetrics, DOWNSTREAM_SASL_AUTHENTICATE_PREDICATE);
+
+            assertThat(saslHandshakeRequestsGoingUpCount)
+                    .isPresent()
+                    .get(DOUBLE)
+                    .withFailMessage("Expecting proxy to have seen at least two handshake requests (one for metadata, one for broker) from downstream")
+                    .isGreaterThanOrEqualTo(2);
+
+            assertThat(saslAuthenticationResponsesComingDown)
+                    .isPresent()
+                    .get(DOUBLE)
+                    .withFailMessage("Expecting proxy to have seen at the same number of authentication responses from the broker as were handshake requests")
+                    .isEqualTo(saslHandshakeRequestsGoingUpCount.get());
+        }
+    }
+
+    @Test
+    void authWithBadToken() throws Exception {
+        var badTokenFile = Files.createTempFile("badtoken", "b64");
+        Files.writeString(badTokenFile, BAD_TOKEN);
+        var config = getClientConfig(badTokenFile.toUri());
```

Comment:
another option for handling the temp file might be:
```suggestion
    void authWithBadToken(@org.junit.jupiter.api.io.TempDir Path tempdir) throws Exception {
        var badTokenFile = Files.createTempFile(tempdir, "badtoken", "b64");
        Files.writeString(badTokenFile, BAD_TOKEN);
```

and let junit recursively delete everything in the tempDir? As In theory `getClientConfig` or `toUri` could throw and leave the temp file around.